From 2dc6ed0ad80bb6380ea2fe09869ae11d97c0e42d Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:33:40 -0400
Subject: [alsa] hda: get subvendor from codec rather than pci_dev

Message-id: <140972823573-314-patchset-jkysela@redhat.com>
Patchwork-id: 91010
O-Subject: [RHEL 7.1 PATCH 313/442] ALSA: hda - get subvendor from codec rather than pci_dev
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

It is safer for non-pci situation.

Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 621b5a047e5c9ced76afd5e8092283032cda802b)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_auto_parser.c b/sound/pci/hda/hda_auto_parser.c
index a142753..b684c6e 100644
--- a/sound/pci/hda/hda_auto_parser.c
+++ b/sound/pci/hda/hda_auto_parser.c
@@ -860,7 +860,7 @@ void snd_hda_pick_pin_fixup(struct hda_codec *codec,
   return;
 
  for (pq = pin_quirk; pq->subvendor; pq++) {
-  if (codec->bus->pci->subsystem_vendor != pq->subvendor)
+  if ((codec->subsystem_id & 0xffff0000) != (pq->subvendor << 16))
    continue;
   if (codec->vendor_id != pq->codec)
    continue;
-- 
1.7.1